app/views/users/registrations/edit.html.erb

```diff
@@ -0,0 +1,53 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
+  <%= devise_error_messages! %>
+
+  <div>
+    <%= f.label :firstname %><br />
```

Comment:
Before you add this to the edit, let's try adding it to the registration form first. `app/views/users/registrations/new.html.erb`